File: testdata/jennies/rawtypes/struct_with_complex_fields/PHPRawTypes/src/Types/StructComplexFields/SomeStruct.php
<?php

namespace Grafana\Foundation\Types\StructComplexFields;

/**
 * This struct does things.
 */
class SomeStruct
{
    public \Grafana\Foundation\Types\StructComplexFields\SomeOtherStruct $fieldRef;

    /**
     * @var string|bool
     */
    public $fieldDisjunctionOfScalars;

    /**
     * @var string|\Grafana\Foundation\Types\StructComplexFields\SomeOtherStruct
     */
    public $fieldMixedDisjunction;

    public ?string $fieldDisjunctionWithNull;

    public \Grafana\Foundation\Types\StructComplexFields\SomeStructOperator $operator;

    /**
     * @var array<string>
     */
    public array $fieldArrayOfStrings;

    /**
     * @var array<string, string>
     */
    public array $fieldMapOfStringToString;

    public \Grafana\Foundation\Types\StructComplexFields\StructComplexFieldsSomeStructFieldAnonymousStruct $fieldAnonymousStruct;

    public string $fieldRefToConstant;
}

File: testdata/jennies/rawtypes/struct_with_complex_fields/CSharpRawTypes/src/Grafana/Foundation/StructComplexFields/SomeStruct.cs
namespace Grafana.Foundation.StructComplexFields;

using System.Collections.Generic;

// This struct does things.
public class SomeStruct
{
    public SomeOtherStruct FieldRef;
    public StringOrBool FieldDisjunctionOfScalars;
    public StringOrSomeOtherStruct FieldMixedDisjunction;
    public string FieldDisjunctionWithNull;
    public SomeStructOperator Operator;
    public List<string> FieldArrayOfStrings;
    public Dictionary<string, string> FieldMapOfStringToString;
    public StructComplexFieldsSomeStructFieldAnonymousStruct FieldAnonymousStruct;
    public string FieldRefToConstant;

    public SomeStruct()
    {
        this.FieldRef = new SomeOtherStruct();
        this.FieldDisjunctionOfScalars = new StringOrBool();
        this.FieldMixedDisjunction = new StringOrSomeOtherStruct();
        this.Operator = SomeStructOperator.GreaterThan;
        this.FieldArrayOfStrings = new List<string>();
        this.FieldMapOfStringToString = new Dictionary<string, string>();
        this.FieldAnonymousStruct = new StructComplexFieldsSomeStructFieldAnonymousStruct();
        this.FieldRefToConstant = new ConnectionPath();
    }

    public SomeStruct(SomeOtherStruct fieldRef, StringOrBool fieldDisjunctionOfScalars, StringOrSomeOtherStruct fieldMixedDisjunction, string fieldDisjunctionWithNull, SomeStructOperator operatorArg, List<string> fieldArrayOfStrings, Dictionary<string, string> fieldMapOfStringToString, StructComplexFieldsSomeStructFieldAnonymousStruct fieldAnonymousStruct, string fieldRefToConstant)
    {
        this.FieldRef = fieldRef;
        this.FieldDisjunctionOfScalars = fieldDisjunctionOfScalars;
        this.FieldMixedDisjunction = fieldMixedDisjunction;
        this.FieldDisjunctionWithNull = fieldDisjunctionWithNull;
        this.Operator = operatorArg;
        this.FieldArrayOfStrings = fieldArrayOfStrings;
        this.FieldMapOfStringToString = fieldMapOfStringToString;
        this.FieldAnonymousStruct = fieldAnonymousStruct;
        this.FieldRefToConstant = fieldRefToConstant;
    }
}
